### PR TITLE
[canary] Adds PyTorch 2.5.1 to canary test

### PR DESCRIPTION
**Pull request by xyang16**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by frankfliu — APPROVED** [review]

File: .github/workflows/canary-aarch64-linux.yml [hunk 1]
```diff
@@ -66,6 +66,8 @@ jobs:
           rm -rf /root/.djl.ai/
           DJL_ENGINE=pytorch-native-auto PYTORCH_VERSION=2.4.0 ./gradlew clean run
```
Comment (frankfliu) [inline .github/workflows/canary-aarch64-linux.yml:67]: We don't really need to support 2.4.0 in 0.31.0

Comment (xyang16) [reply .github/workflows/canary-aarch64-linux.yml:67]: 2.4.0 is published in nightly_publish, should we remove it from nightly_publish first?

Comment (xyang16) [reply .github/workflows/canary-aarch64-linux.yml:67]: I think we can remove 1.13.1, since it's pretty old.

Comment (frankfliu) [reply .github/workflows/canary-aarch64-linux.yml:67]: We need keep 1.13.1, that for cuda11 and last version support AL2
We also need keep 2.3.1 that for cu121

Comment (xyang16) [reply .github/workflows/canary-aarch64-linux.yml:67]: ok

Comment (xyang16) [reply .github/workflows/canary-aarch64-linux.yml:67]: We will remove it in the next release.

**Review by siddvenk — APPROVED** [review]

**Pull request merged**